docs/abi-spec.rst

```diff
@@ -191,7 +191,7 @@ on the type of ``X`` being
 
 - ``T[]`` where ``X`` has ``k`` elements (``k`` is assumed to be of type ``uint256``):
 
-  ``enc(X) = enc(k) enc([X[0], ..., X[k-1]])``
+  ``enc(X) = enc(k) enc(([X[0], ..., X[k-1]]))``
```

Comment:
Without looking at it and thinking through it in all detail, I actually think the previous version is correct and the change *makes* this incorrect.
``[X[0], ..., X[k-1]]`` is meant to be a fixed size array, which will then recurse back to the previous clause, ultimately resulting in ``(X[0], ..., X[k-1])``.

A single-element tuple of a statically-sized array, i.e. the version after the change, can actually be encoded differently (e.g. in case the base type is dynamic).

Reply:
It would be an option to change this to ``enc((X[0], ..., X[k-1]))`` (note the missing ``[ ]`` compared to the version in the PR).
But if we did that, we'd also need to change the comment below.

Here one could argue either way - to me it seems nicer to reduce dynamic arrays to static arrays first, and only then to tuples, rather than directly to tuples - but we don't actually formally introduce the ``[...]`` notation for static arrays, do we? If we don't formally introduce that notation, directly moving to a tuple may indeed be simpler.

Reply:
My bad. I meant it to match `enc((X[0], ..., X[k-1]))` above but somehow failed to remove `[]` brackets.

Reply:
@ekpyron PTAL